Weapon Changes

Metroid55 opened the pull request:
Trying to fix the op weapons and get some damage out of the game

Metroid55 committed:
Weapon Changes

Trying to fix the op weapons and get some damage out of the game

swampservers commented on lua/cinema/guns/weapons/gun/shared.lua:
```diff
@@ -155,20 +155,22 @@ function GunPerkOverrides(swep, perk)
         t.UnscopeOnShoot = false
     elseif perk == "boomstick" then
-        t.NumPellets = swep.NumPellets * 2
+        t.NumPellets = swep.NumPellets * 1.5
```
this is too much nerf, since the spread is increased by 1.5x its gonna suck

swampservers commented:
I'm not sure about all these changes that reduce the base damage by 1, this will result in a lot more almost-kills where someone ends with just a few % health left. Did you test this at all or just putting some values in? I'm fine with nerfing perks like boomstick if they're OP

Metroid55 committed:
boomstick spread + devtools

- adjusted boomstick spread to make it less "sucky"
- made slug slighty cheaper with the nerfs
- added id to devtools so I can test my changes if needed

Metroid55 commented:
The problem is that basically almost every gun is way too cheap and kills too fast. I'm trying to negate some of that, because there is no armor like in css and perks like armor piercing exist to make your weapon more deadly (but no one uses it when the base weapon kills perfectly fine and is cheaper). Most of the killing happens at a close range and weapons like the shotguns, smgs and pistols either 1-shot you (even without perks) or kill you in a split second. I would even go as far as too say that my changes are quite tame and if I could I would increase the price of some weapons/weaponclasses as well (increase pistol prices by 30-40% (more for the deagle and magnum), shotguns by 10%, smgs by 20%, ...)

swampservers commented:
Maybe we should make it so armor piercing just gives the current default damage values, and non armor piercing divides the damage by what the AP multiplier used to be? That would make all the guns quite a bit weaker. I'm also considering headshot damage and which guns should be a 1 shot headshot.

Metroid55 commented:
Some guns might be a bit too weak with that change (famas for example is already a joke gun + also snipers would feel weird), but we could try that for now and adjust accordingly. You would have to do that though, because I don't know how.

Metroid55 committed:
Adjusted with dmg nerfs in mind

Metroid55 committed:
Update gun_mp7.lua

Metroid55 committed:
Update shared.lua

Metroid55 commented:
Updated my pr with the upcoming changes in mind

Metroid55 committed:
don't need the price increase on boomstick

Metroid55 committed:
oops

Metroid55 committed:
revert slug range change

LegacyValken commented:
> The problem is that basically almost every gun is way too cheap and kills too fast. I'm trying to negate some of that, because there is no armor like in css and perks like armor piercing exist to make your weapon more deadly (but no one uses it when the base weapon kills perfectly fine and is cheaper). Most of the killing happens at a close range and weapons like the shotguns, smgs and pistols either 1-shot you (even without perks) or kill you in a split second. I would even go as far as too say that my changes are quite tame and if I could I would increase the price of some weapons/weaponclasses as well (increase pistol prices by 30-40% (more for the deagle and magnum), shotguns by 10%, smgs by 20%, ...)

If you want to fix the cheap problem then airsoft weapons need to have their own separate ammo type so players don't just buy airsoft ammo at 100 to 300 points per round when refill cost is meant to be 1.5k to 3k for epics. The cost of constant RDM should eat a bit into profits.

PuraPoint commented:
> 

Thats a bug, not a feature

swampservers commented:
So here's the thing. I don't want to merge this because I carefully selected the guns' damage values and dropoff to have a certain number of shots to kill at different ranges. I don't want to lose that.

I think a better way to do this would be to, if the gun does NOT have the armor piercing perk, modify the damage before it is applied so that there is 1 more shot to kill, so to speak.

A quick solution would be something like, on gun/shared.lua line 92, do something like this:
```
local damage = swep.Damage

if swep.GunType == "smg" then
    damage = 100/( (100/damage)+1 )
end
```
then replace all uses of swep.Damage with damage in the function.

This way, all SMGs take 1 more shot to kill.

Metroid55 committed:
damage

Metroid55 committed:
fiveseven damage (see m4a1)

Metroid55 committed:
Update gun_spas12.lua

Metroid55 commented:
Okay I updated it. If you are okay with those changes I can create a new pull request if you want, so that there aren't that many commits.

Metroid55 committed:
compensation to make the ump not garbage

Metroid55 committed:
better I think

Metroid55 committed:
airsoft fix

Metroid55 commented:
Added airsoft fix.
